packages/core-contracts/src/contracts/arks/SiUSDArk.sol

```diff
@@ -0,0 +1,190 @@
+        siUSD = IERC4626(_siUSD);
+
+        // Approve Gateway to spend USDC for mintAndStake operations
+        config.asset.forceApprove(_gateway, Constants.MAX_UINT256);
+
+        // Approve Gateway to spend siUSD for unstake operations
+        IERC20(_siUSD).forceApprove(_gateway, Constants.MAX_UINT256);
+
+        // Approve Gateway to spend iUSD (siUSD's asset) for redeem operations
+        IERC20(siUSD.asset()).forceApprove(_gateway, Constants.MAX_UINT256);
+    }
+
+    /*//////////////////////////////////////////////////////////////
+                            VIEW FUNCTIONS
+    //////////////////////////////////////////////////////////////*/
+
+    /**
+     * @inheritdoc IArk
+     * @notice Returns the total assets managed by this Ark, denominated in USDC
+     * @dev Converts siUSD shares -> iUSD (via ERC4626)
+     *      Note: iUSD is approximately 1:1 with USDC (adjusted for decimals)
+     *      iUSD has 18 decimals, USDC has 6 decimals
+     * @return assets The total balance of USDC-equivalent assets
+     */
+    function totalAssets() public view override returns (uint256 assets) {
+        uint256 siUSDShares = siUSD.balanceOf(address(this));
+        if (siUSDShares > 0) {
+            // Convert siUSD shares to iUSD amount (18 decimals)
+            uint256 iUSDAmount = siUSD.convertToAssets(siUSDShares);
+            // Convert iUSD (18 decimals) to USDC equivalent (6 decimals)
+            // iUSD is pegged 1:1 to USD, so we just adjust decimals
+            assets = iUSDAmount / 1e12;
+        }
+    }
+
+    /*//////////////////////////////////////////////////////////////
+                            INTERNAL FUNCTIONS
+    //////////////////////////////////////////////////////////////*/
+
+    /**
+     * @notice Internal function to get the total assets that are withdrawable
+     * @dev Returns the USDC-equivalent value of assets that can be withdrawn
+     * @return withdrawableAssets Amount of USDC that can be withdrawn
+     */
+    function _withdrawableTotalAssets()
+        internal
+        view
+        override
+        returns (uint256 withdrawableAssets)
+    {
+        uint256 siUSDShares = siUSD.balanceOf(address(this));
+        if (siUSDShares > 0) {
+            // Get maximum iUSD we can withdraw from siUSD vault (18 decimals)
+            uint256 maxIUSD = siUSD.maxWithdraw(address(this));
+            // Convert to USDC equivalent (6 decimals)
+            withdrawableAssets = maxIUSD / 1e12;
+        }
+    }
+
+    /**
+     * @notice Deposits USDC into InfiniFi via Gateway
+     * @dev Uses Gateway.mintAndStake() to convert USDC -> siUSD directly
+     * @param amount The amount of USDC to deposit
+     */
+    function _board(uint256 amount, bytes calldata) internal override {
+        // Use Gateway to mint iUSD and stake to siUSD in one transaction
+        // This returns the amount of iUSD that was minted (and staked)
+        gateway.mintAndStake(address(this), amount);
```

Comment:
Don't we want to validate that the minted/staked amount is within reasonable numbers? I know it is not going to be 1:1, but perhaps a static slippage check here would catch errors in the InfiniFi protocol

Reply:
My take is this:
- the USDC -> iUSD peg is enforced by the gateway. 
- the iUSD -> SiUSD is enforced by the derministic maths of the ERC4626 staking vault

So, I'm not sure we get anything from slippage checks here.